modules/data.land/R/IC_SOILGRID_Utilities.R

```diff
@@ -0,0 +1,373 @@
+    still_missing <- is.na(processed$`Std_soilC_0-30cm`)
+    if (any(still_missing)) {
+      processed$`Std_soilC_0-30cm`[still_missing] <- 
+        processed$`Total_soilC_0-30cm`[still_missing] * default_uncertainty
+      PEcAn.logger::logger.warn(sprintf(
+        "Using default uncertainty (%.1f%% of mean) for %d site(s)", 
+        default_uncertainty * 100, sum(still_missing)
+      ))
+      
+      if (verbose) {
+        for (i in which(still_missing)) {
+          PEcAn.logger::logger.debug(sprintf(
+            "Using default uncertainty (%.1f%% of mean) for site %s", 
+            default_uncertainty * 100, processed$Site_ID[i]
+          ))
+        }
+      }
+    }
+  }
+  
+  # Ensure standard deviation is non-negative 
+  neg_sd_count <- sum(processed$`Std_soilC_0-30cm` < 0, na.rm = TRUE)
+  if (neg_sd_count > 0) {
+    PEcAn.logger::logger.warn(sprintf("Found %d negative standard deviations", neg_sd_count))
+    processed$`Std_soilC_0-30cm` <- pmax(processed$`Std_soilC_0-30cm`, 0, na.rm = TRUE)
+  }
+  
+  # Ensure mean is non-negative 
+  neg_mean_count <- sum(processed$`Total_soilC_0-30cm` < 0, na.rm = TRUE)
+  if (neg_mean_count > 0) {
+    PEcAn.logger::logger.warn(sprintf("Found %d negative mean values", neg_mean_count))
+    processed$`Total_soilC_0-30cm` <- pmax(processed$`Total_soilC_0-30cm`, 0, na.rm = TRUE)
+  }
+  
+  # Add minimum standard deviation to avoid zero uncertainty 
+  min_sd <- 0.1 * processed$`Total_soilC_0-30cm` # 10% of mean as minimum SD
+  is_zero_sd <- processed$`Std_soilC_0-30cm` == 0 | is.na(processed$`Std_soilC_0-30cm`)
+  zero_sd_count <- sum(is_zero_sd)
+  
+  if (zero_sd_count > 0) {
+    PEcAn.logger::logger.info(sprintf("Setting minimum uncertainty for %d zero/NA standard deviations", 
+                                     zero_sd_count))
+    processed$`Std_soilC_0-30cm` <- pmax(processed$`Std_soilC_0-30cm`, min_sd, na.rm = TRUE)
+  }
+  
+  return(processed)
+}
+
+#' Generate ensemble members for a site
+#' 
+#' @param processed_data Processed soil carbon data
+#' @param site_id Site ID
+#' @param lat Site latitude
+#' @param lon Site longitude
+#' @param ensemble_size Number of ensemble members to create
+#' @param verbose Print detailed progress information to the terminal? TRUE/FALSE
+#' 
+#' @return List of ensemble data for the site
+#' @export
+generate_soilgrids_ensemble <- function(processed_data, site_id, lat, lon, ensemble_size, verbose = FALSE) {
+  if (verbose) {
+    PEcAn.logger::logger.info(sprintf("Generating %d ensemble members for site %s", ensemble_size, site_id))
+  }
+  
+  # Get site row from processed data
+  site_row <- which(processed_data$Site_ID == site_id)
+  if (length(site_row) == 0) {
+    PEcAn.logger::logger.severe(sprintf("Site %s not found in processed data", site_id))
+  }
+  
+  # Set random seed for reproducibility
+  set.seed(as.numeric(site_id))
+  
+  # Generate all ensemble members at once 
+  soil_c_values <- truncnorm::rtruncnorm(
```

Comment:
Unclear why a truncated normal is being assumed when all prior analyses conducted by @Qianyuxuan suggested a gamma